REG_RSP is an enum value in the NDK so cannot be ifdef checked.

diff -u -r ../ruby-2.3.1/signal.c ./signal.c
--- ../ruby-2.3.1/signal.c	2015-11-23 19:17:11.000000000 -0500
+++ ./signal.c	2016-06-17 17:15:18.947494468 -0400
@@ -771,7 +771,7 @@
 {
     const DEFINE_MCONTEXT_PTR(mctx, ctx);
 # if defined __linux__
-#   if defined REG_RSP
+#   if defined __x86_64__
     const greg_t sp = mctx->gregs[REG_RSP];
 #   else
     const greg_t sp = mctx->gregs[REG_ESP];
